### PR TITLE
TRIFFHIR-543: When a structuredefintion saves, IG's reference to the …

### PR DESCRIPTION
**Pull request by ajagann1**
…structuredefinition is updated to reflect the title if it exists or the name if it doesn't.

### CODE REVIEW

File: apps/client/src/app/structure-definition/structure-definition.component.ts [hunk 1]
```diff
@@ -300,6 +302,22 @@ export class StructureDefinitionComponent extends BaseComponent implements OnIni
...
+
+    const implementationGuideId = this.route.snapshot.paramMap.get('implementationGuideId');
+    const implementationGuide = await this.implementationGuideService.getImplementationGuide(implementationGuideId).toPromise();
+    const resources = (<ImplementationGuide> implementationGuide).definition.resource;
+
+
+    const index = resources.findIndex(resource => {
+      return resource.reference.reference.indexOf(this.structureDefinition.id) > 0;
+    });
+    if(index === -1) throw new Error(`Couldn't find structure definition in resources`);
```
Comment (seanmcilvenna) [inline apps/client/src/app/structure-definition/structure-definition.component.ts:314]: Don't throw an error here. Just wrap the next block with
`if (index >= 0) { ... }`

File: apps/client/src/app/structure-definition/structure-definition.component.ts [hunk 2]
```diff
@@ -300,6 +302,22 @@ export class StructureDefinitionComponent extends BaseComponent implements OnIni
...
+
+    (<ImplementationGuide> implementationGuide).definition.resource[index].name =
+      this.structureDefinition.title ? this.structureDefinition.title : this.structureDefinition.name;
+
+    await this.implementationGuideService.saveImplementationGuide(<ImplementationGuide> implementationGuide)
+      .subscribe( () => {});
```
Comment (seanmcilvenna) [inline apps/client/src/app/structure-definition/structure-definition.component.ts:320]: Don't think `.susbcribe()` is needed here if you're not doing anything with it.
Might consider something like this instead:
```
await this.implementationGuideService.saveImplementationGuide(<ImplementationGuide> implementationGuide)
  .toPromise()
  .catch(err => console.log(err));
```

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]
In general, approach is fine. However, in future, might consider whether logic like this could be better served on the server, so that if people (for whatever reason) use the ToF API instead of the UI, this logic would still be applied. No need to change now, though. Let's leave it as is, cause it works, and move on for now.

File: apps/client/src/app/structure-definition/structure-definition.component.ts (hunk 2 above)
Comment (ajagann1) [reply apps/client/src/app/structure-definition/structure-definition.component.ts]: Added the requested changes.  Let me know if you think anything else is needed

**Review by seanmcilvenna — APPROVED** [review]

**Pull request merged**